miden-tx/src/host/account_delta.rs

```diff
@@ -169,9 +169,11 @@ impl<A: AdviceProvider> TransactionHost<A> {
 ///
 /// The delta tracker is composed of:
 /// - A map which records the latest states for the updated storage slots.
+/// - A map which records the latest states for the updates storage maps
 #[derive(Default, Debug, Clone, PartialEq, Eq)]
 struct AccountStorageDeltaTracker {
```

Comment:
@bobbinth @hackaugusto here it is not clear to me if I use the `AccountStorageDeltaTracker` correctly. The tests pass, but I am not sure if we test the right things here.

Reply:
I changed it, but I am still unsure where we populate the `AccountStorageDeltaTracker`. So I am pretty sure the implementation is incorrect atm.

Reply:
It should be populated [here](https://github.com/0xPolygonMiden/miden-base/blob/main/miden-tx/src/host/account_delta.rs#L88). We'll need to add a new event for storage map updates. This event would be processed in the host to figure out what exactly is being updated and record the updates accordingly.